src/scanner.c

```diff
@@ -3091,8 +3110,7 @@ static Int InitKernel (
     Int                 i;
 
     TLS(Input) = TLS(InputFiles);
-    TLS(Input)--;
-    (void)OpenInput(  "*stdin*"  );
+    (void)OpenInput_Internal(  "*stdin*", 1  );
```

Comment:
Nope, your change is fine. I will reduce this to the other, tiny commit in this patch.